modules/select/js/select_directive.js

```diff
@@ -216,6 +216,22 @@ angular.module('lumx.select', [])
             }
         }
 
+        /**
+         * Filter using the default filter only if a custom filter is not defined
+         */
+        $scope.filterFn = function(array)
+        {
+            if ($scope.filter)
+            {
+                return array;
+            }
+            else
+            {
+                return $filter('filter')(array, $scope.data.filter);
```

Comment:
It returns always a new filtered array, it might create infinite digest loop sometimes.